uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/EntitlementCalculation.php

```diff
@@ -0,0 +1,406 @@
+   * period.
+   *
+   * The Pro Rata is given by:
+   * (no. working days to work / no. of working days) x contractual entitlement.
+   *
+   * The end result is rounded up to the nearest half day. Example:
+   *
+   * Number of working days to work: 212
+   * Number of working days: 253
+   * Contractual entitlement: 28
+   * Pro rata: (212 / 253) * 28 = 23.46 = 23.5 (rounded)
+   *
+   * @return float|int
+   */
+  public function getProRata() {
+    $numberOfWorkingDaysToWork = $this->getNumberOfWorkingDaysToWork();
+    $numberOfWorkingDays = $this->getNumberOfWorkingDays();
+    $contractualEntitlement = $this->getContractualEntitlement();
+
+    $proRata = ($numberOfWorkingDaysToWork / $numberOfWorkingDays) * $contractualEntitlement;
+    $roundedProRata = ceil($proRata * 2) / 2;
+
+    return $roundedProRata;
+  }
+
+  /**
+   * Returns the Contractual Entitlement for this calculation contract.
+   *
+   * The Contractual Entitlement is given by the leave amount set for this
+   * calculation contract and absence type + the number of public holidays in
+   * the period (if the leave settings on the contract allows this).
+   *
+   * @return int
+   */
+  public function getContractualEntitlement() {
+    $contractualEntitlement = 0;
+
+    $jobLeave = $this->getJobLeaveForAbsenceType();
+    if(!$jobLeave) {
+      return $contractualEntitlement;
+    }
+
+    $contractualEntitlement = $jobLeave['leave_amount'];
+    if($jobLeave['add_public_holidays']) {
+      $contractualEntitlement += $this->getNumberOfPublicHolidaysForPeriod();
+    }
+
+    return $contractualEntitlement;
+  }
+
+  /**
+   * Returns the calculated proposed entitlement.
+   *
+   * This is basically the Pro Rata + the number of days brought forward
+   *
+   * @return float|int
+   */
+  public function getProposedEntitlement() {
+    return $this->getProRata() + $this->getBroughtForward();
+  }
+
+  /**
+   * Returns the proposed entitlement for this AbsenceType and Contract on the
+   * previous period.
+   *
+   * @return int
+   */
+  public function getPreviousPeriodProposedEntitlement() {
+    $previousPeriodEntitlement = $this->getPreviousPeriodEntitlement();
+
+    if(!$previousPeriodEntitlement) {
+      return 0;
+    }
+
+    return $previousPeriodEntitlement->proposed_entitlement;
+  }
+
+  /**
+   * Return the number of Leaves taken during the Previous Period
+   *
+   * @TODO The Leaves Request feature is not yet implemented, so we're only returning 0 for now
+   *
+   * @return int
+   */
+  public function getNumberOfLeavesTakenOnThePreviousPeriod() {
+    return 0;
+  }
+
+  /**
+   * Return the number of days remaining on the previous period. That is,
+   * the proposed entitlement - the number of leaves taken
+   *
+   * @return int
+   */
+  public function getNumberOfDaysRemainingInThePreviousPeriod() {
+    $entitlement = $this->getPreviousPeriodEntitlement();
+
+    if(!$entitlement) {
```

Comment:
I would convert this to the short if syntax like this :

return !$entitlement : 0 ? $entitlement->getNumberOfDaysRemaining();